roles/splunk/tasks/kvstore_upgrade.yml

```diff
@@ -0,0 +1,156 @@
+---
+- name: Check if authenticated
+  include_tasks: splunk_login.yml
+  when: not splunk_authenticated
+
+- name: Set fact for kvstore Upgrade
+  ansible.builtin.set_fact:
+    splunk_upgrade_server: false
+
+- name: Check the current kvstore storage backend
+  ansible.builtin.command: |
+    {{ splunk_home }}/bin/splunk show kvstore-status | grep storageEngine | sed -r 's/\s+storageEngine : //g'
+  become: true
+  become_user: "{{ splunk_nix_user }}"
+  register: splunk_kvstore_backend_out
+  changed_when: splunk_kvstore_backend_out.rc == 0
```

Comment:
I personally like to set `changed_when: false` for checks. In my head it makes sense.

Reply:
fair play

Reply:
Fixed in 3263c35